DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/RootBaseObjectBuilder.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Rhino.DocObjects;
+using Rhino;
+using Speckle.Connectors.DUI.Bindings;
+using Speckle.Core.Models;
+using Speckle.Core.Kits;
+using System.Threading;
+
+namespace Speckle.Connectors.Rhino7.Operations.Send;
+
+/// <summary>
+/// Stateless builder object to turn an <see cref="ISendFilter"/> into a <see cref="Base"/> object
+/// </summary>
+public class RootBaseObjectBuilder
+{
+  private readonly Func<ISpeckleConverter> _converterFactory;
+
+  public RootBaseObjectBuilder(Func<ISpeckleConverter> converterFactory)
```

Comment:
Do we need a factory here? Can't we just resolve the ISpeckleConverter directly?

Reply:
This does need to be a factory here with the current converter. This is because the rootBaseObjectBuilder is designed to be a stateless singleton that is reused for all send operations. Therefore, it can't reuse the same converter instance because our converter is full of state.

If the new converter is being designed to be stateless then we could obviously change this